File: AN_2/SEM_2/MPP/java/lab4/src/main/java/ro/mpp/labfx/domain/Utilizator.java
package ro.mpp.labfx.domain;

import org.mindrot.jbcrypt.BCrypt;

import java.util.Objects;

public class Utilizator implements Identifiable<Integer>{
    private int idUtilizator;
    private String nume;
    private String parola;

    public Utilizator(String nume, String parola) {
        this.nume = nume;
        setParola(parola);
    }

    public Utilizator(int idUtilizator,String nume, String parola) {
        this.idUtilizator = idUtilizator;
        this.nume = nume;
        this.parola = parola;
    }

    @Override
    public void setId(Integer integer) {
        this.idUtilizator = integer;
    }

    @Override
    public Integer getId() {
        return idUtilizator;
    }

    public String getNume() {
        return nume;
    }

    public void setNume(String nume) {
        this.nume = nume;
    }

    public String getParola() {
        return parola;
    }

    public void setParola(String parola) {
        this.parola = BCrypt.hashpw(parola, BCrypt.gensalt());
    }

    public boolean verificaParola(String parola) {
        return BCrypt.checkpw(parola, this.parola);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Utilizator utilizator = (Utilizator) o;
        return idUtilizator == utilizator.idUtilizator;
    }

    @Override
    public int hashCode() {
        return Objects.hash(idUtilizator);
    }

    @Override
    public String toString() {
        return "Utilizator{" +
                "idUtilizator=" + idUtilizator +
                ", nume='" + nume + '\'' +
                ", parola='" + parola + '\'' +
                '}';
    }
}

File: AN_2/SEM_2/MPP/c#/lab3/domain/Utilizator.cs
namespace lab3.domain;
public class Utilizator : Identifiable<int>
{
    private int idUtilizator;
    private string nume;
    private string parola;

    public Utilizator(string nume, string parola)
    {
        this.nume = nume;
        this.parola = parola;
    }

    public Utilizator(int idUtilizator, string nume, string parola)
    {
        this.idUtilizator = idUtilizator;
        this.nume = nume;
        this.parola = parola;
    }

    public void SetId(int id) => this.idUtilizator = id;
    public int GetId() => idUtilizator;

    public string GetNume() => nume;
    public void SetNume(string nume) => this.nume = nume;

    public string GetParola() => parola;
    public void SetParola(string parola) => this.parola = parola;

    public override bool Equals(object? obj)
    {
        if (this == obj) return true;
        if (obj == null || GetType() != obj.GetType()) return false;
        var utilizator = (Utilizator)obj;
        return idUtilizator == utilizator.idUtilizator;
    }

    public override int GetHashCode() => idUtilizator.GetHashCode();

    public override string ToString()
    {
        return $"Utilizator{{idUtilizator={idUtilizator}, nume='{nume}', parola='{parola}'}}";
    }
}
